packages/aws-utils/bin/bootstrapAppSyncDynamo.js

```diff
@@ -0,0 +1,310 @@
+                `arn:aws:dynamodb:${config.region}:*:table/${entity.name}/*`,
+              ], []),
+          }],
+        },
+      }],
+    },
+  };
+
+  return resources;
+};
+
+const writeGraphqlSchema = (dir) => {
+  console.log('Ѱζ༼ᴼل͜ᴼ༽ᶘѰ    Writing schema.graphql');
+  const { entities } = config;
+  const writeStream = fs.createWriteStream(`${dir}/schema.graphql`);
+
+  return new Promise((resolve) => {
+    writeStream.on('error', (e) => {
+      console.error(e);
+      process.exit(1);
+    });
+
+    writeStream.on('finish', resolve);
+
+    const writeInput = type => writeStream.write(`\ninput ${type} {\n`);
+    const writeType = type => writeStream.write(`\ntype ${type} {\n`);
+    const writeEnd = () => writeStream.write('}\n');
+
+    writeStream.write('schema {\n');
+    writeStream.write('  query: Query\n');
+    writeStream.write('  mutation: Mutation\n');
+    if (entities.some(entity => entity.subscriptions)) {
+      writeStream.write('  subscription: Subscription\n');
+    }
+    writeEnd();
+
+    writeType('Query');
+    entities.forEach((entity) => {
+      const type = entity.name;
+      writeStream.write(`  ${type}ById(id: ID!): ${type}\n`);
+      writeStream.write(`  ${type}(query: [AttributeFilter]): [${type}]!\n`);
+    });
+    writeEnd();
+
+    writeType('Mutation');
+    entities.forEach((entity) => {
+      const { name } = entity;
+      writeStream.write(`  put${name}(id: ID, input: ${name}Input!): ${name}!\n`);
+      writeStream.write(`  update${name}(id: ID!, input: ${name}Input!): ${name}!\n`);
+      writeStream.write(`  delete${name}(id: ID!): Boolean\n`);
+    });
+    writeEnd();
+
+    if (entities.some(entity => entity.subscriptions)) {
+      writeType('Subscription');
+
+      entities
+        .filter(entity => entity.subscriptions)
+        .forEach((entity) => {
+          const { name } = entity;
+          entity.subscriptions.forEach((subscription) => {
+            writeStream.write(`  subscribeTo${capitalize(subscription)}${name}: ${name}!\n`);
+            writeStream.write(`  @aws_subscribe(mutations: ["${subscription}${name}"])\n`);
+          });
+        });
+      writeEnd();
+    }
+
+    entities.forEach((entity) => {
+      const { attributes, name } = entity;
+
+      writeType(name);
+      writeStream.write('  id: ID!\n');
+      writeStream.write(`${attributes.map(attribute => `  ${attribute.name}: ${attribute.type}`).join('\n')}\n`);
+      writeEnd();
+
+      writeInput(`${name}Input`);
+      writeStream.write(`${attributes.map(attribute => `  ${attribute.name}: ${attribute.type}`).join('\n')}\n`);
+      writeEnd();
+    });
+
+    writeInput('AttributeFilter', true);
+    writeStream.write('  expression: String!\n');
+    writeStream.write('  expressionName: String!\n');
+    writeStream.write('  expressionNumberValue: Float\n');
+    writeStream.write('  expressionStringValue: Float\n');
+    writeEnd();
```

Comment:
General comment here... I think this would be a lot more readable as a template string. Also you could use fs.writeSync here easily as you are not trying to handle a bunch of parallel (waiting) IO.